subsystems/provisioning/provisioning.go

```diff
@@ -372,27 +364,34 @@ func (w *Provisioning) updateHotspotSSID(cfg *Config) {
 	}
 }
 
-func (w *Provisioning) writeWifiPowerSave() error {
+// must be run inside dataMu lock.
+func (w *Provisioning) writeWifiPowerSave(ctx context.Context) error {
 	contents := wifiPowerSaveContentsDefault
-
-	if w.cfg.DisableWifiPowerSave != nil && *w.cfg.DisableWifiPowerSave {
-		contents = wifiPowerSaveContentsDisable
-	}
-
-	if w.cfg.DisableWifiPowerSave != nil && !*w.cfg.DisableWifiPowerSave {
-		contents = wifiPowerSaveContentsEnable
+	if w.cfg.WifiPowerSave != nil {
+		if *w.cfg.WifiPowerSave {
+			contents = wifiPowerSaveContentsEnable
+		} else {
+			contents = wifiPowerSaveContentsDisable
+		}
 	}
 
 	isNew, err := agent.WriteFileIfNew(wifiPowerSaveFilepath, []byte(contents))
 	if err != nil {
-		return errw.Wrap(err, "error writing wifi-powersave.conf")
+		return errw.Wrap(err, "writing wifi-powersave.conf")
 	}
 
 	if isNew {
 		w.logger.Infof("Updated %s to: %q", wifiPowerSaveFilepath, contents)
 		// Reload NetworkManager to apply changes
 		if err := w.nm.Reload(0); err != nil {
-			return errw.Wrap(err, "error reloading NetworkManager after wifi-powersave.conf update")
+			return errw.Wrap(err, "reloading NetworkManager after wifi-powersave.conf update")
+		}
+
+		ssid := w.netState.ActiveSSID(w.cfg.HotspotInterface)
+		if w.connState.getConnected() && ssid != "" {
```

Comment:
i guess we're checking with only do this if we're not in hotspot mode which is a good catch